src/main/java/seedu/address/logic/commands/PrintCommand.java

```diff
@@ -0,0 +1,92 @@
+
+import seedu.address.model.insurance.ReadOnlyInsurance;
+import seedu.address.model.insurance.UniqueLifeInsuranceList;
+import seedu.address.model.person.ReadOnlyPerson;
+
+/**
+ * Prints the list of contacts, along with any associated
+ * insurance policies where the contact is involved in,
+ * into a printable, readable .txt file.
+ */
+public class PrintCommand extends Command {
+
+    public static final String[] COMMAND_WORDS = {"print"};
+    public static final String COMMAND_WORD = "print";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Saves the addressbook into a .txt file named by you for your viewing.\n"
+            + "Example: " + COMMAND_WORD + " filename\n"
+            + "file can then be found in the in doc/books folder as filename.txt";
+
+    public static final String MESSAGE_SUCCESS = "Address Book has been saved!\n"
+            + "Find your Address Book in the %1$s.txt file you created in the doc/books folder.";
+
+    private final String fileName;
+
+    public PrintCommand(String filename) {
+        requireNonNull(filename);
+
+        this.fileName = filename;
+    }
+
+
+    @Override
+    public CommandResult execute() {
+
+        List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
+
+        List<String> lines = new ArrayList<>();
+        String timeStamp = new SimpleDateFormat("dd/MM/YYYY" + " " + "HH:mm:ss").format(new Date());
+        lines.add("Addressbook was last updated on: " + timeStamp + "\n");
+
+        int personIndex = 1;
+        for (ReadOnlyPerson person: lastShownList) {
+            String entry = personIndex + ". " + person.getAsText();
+            lines.add(entry);
+
+            UniqueLifeInsuranceList insurances = person.getLifeInsurances();
+            for (ReadOnlyInsurance insurance: insurances) {
+                lines.add("Insurance Policy: =========");
+                String owner = insurance.getOwner().getName();
+                String insured = insurance.getInsured().getName();
+                String beneficiary = insurance.getBeneficiary().getName();
+                String premium = insurance.getPremium().toString();
+                String signingDate = insurance.getSigningDateString();
+                String expiryDate = insurance.getExpiryDateString();
+                lines.add("Owner: " + owner + "\n"
+                        + "Insured: " + insured + "\n"
+                        + "Beneficiary: " + beneficiary + "\n"
+                        + "Premium: " + premium + "\n"
+                        + "Signing Date: " + signingDate + "\n"
+                        + "Expiry Date: " + expiryDate + "\n"
+                );
+                lines.add("============");
+            }
+            personIndex++;
+        }
+
+        Path file = Paths.get("docs/books/" + fileName + ".txt");
+        try {
+            Files.write(file, lines, Charset.forName("UTF-8"));
+        } catch (IOException e) {
```

Comment:
by convention we just follow the exception in naming, this should be ioe

Reply:
Fixed